hathor/manager.py

```diff
@@ -703,6 +710,35 @@ def generate_parent_txs(self, timestamp: Optional[float]) -> 'ParentTxs':
         can_include = [i.data for i in can_include_intervals]
         return ParentTxs(max_timestamp, can_include, must_include)
 
+    def get_best_blockchain(self, n_blocks: int) -> list['BlockInfo']:
+        """ Get a list with N blocks from the best blockchain
+        in descending order.
+        """
+        if not (n_blocks > 0):
+            raise ValueError(
+                'Invalid N. N must be greater than 0.'
+            )
+
+        block = self.tx_storage.get_best_block()
+        if self._latest_best_blockchain:
+            best_blockchain_tip = self._latest_best_blockchain[0]
+            if block.to_blockinfo() == best_blockchain_tip:
+                return self._latest_best_blockchain
+
+        best_blockchain: list[BlockInfo] = []
+        while len(best_blockchain) < n_blocks:
+            block_info = BlockInfo(
+                block.hash_hex,
+                block.get_height(),
+                block.weight)
+            best_blockchain.append(block_info)
+            if block.is_genesis:
+                break
+            block = block.get_block_parent()
+
+        self._latest_best_blockchain = best_blockchain
```

Comment:
A small observation: this cache isn't guaranteed to have a consistent length (calls with different `n_blocks` will lead to caches of different sizes). I don't thinks this is really an issue _here_, but it might be worth adding a comment saying that this isn't a problem.